source/slang-capture-replay/slang-type-conformance.cpp

```diff
@@ -0,0 +1,131 @@
+#include "capture_utility.h"
+#include "slang-type-conformance.h"
+
+namespace SlangCapture
+{
+    TypeConformanceCapture::TypeConformanceCapture(slang::ITypeConformance* typeConformance)
+        : m_actualTypeConformance(typeConformance)
+    {
+        assert(m_actualTypeConformance != nullptr);
+        slangCaptureLog(LogLevel::Verbose, "%s: %p\n", __PRETTY_FUNCTION__, m_actualTypeConformance);
+    }
+    TypeConformanceCapture::~TypeConformanceCapture()
+    {
+        m_actualTypeConformance->release();
+    }
+
+    ISlangUnknown* TypeConformanceCapture::getInterface(const Guid& guid)
+    {
+        if (guid == TypeConformanceCapture::getTypeGuid())
+        {
+            return static_cast<ISlangUnknown*>(this);
+        }
+        else
+        {
+            return nullptr;
+        }
+    }
+
+    SLANG_NO_THROW slang::ISession* TypeConformanceCapture::getSession()
+	{
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+        slang::ISession* res = m_actualTypeConformance->getSession();
+        return res;
+    }
+
+    SLANG_NO_THROW slang::ProgramLayout* TypeConformanceCapture::getLayout(
+        SlangInt    targetIndex,
+        slang::IBlob**     outDiagnostics)
+	{
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+        slang::ProgramLayout* res = m_actualTypeConformance->getLayout(targetIndex, outDiagnostics);
+        return res;
+    }
+
+    SLANG_NO_THROW SlangInt TypeConformanceCapture::getSpecializationParamCount()
+	{
```

Comment:
I see that the indentation problem is when a tab-character is used.
Please replace tab-characters with 4 space-characters.

Reply:
The tab thing is because I use "%s//" in vim to replace something globally. But I forgot using `:retab`, Now it's OK.